### ISSUE TITLE
False positive consider-using-with on a `with` statement

### ISSUE THREAD

**Issue by adam-azarchs** [PyCQA/pylint#4676]
### Steps to reproduce

Given a file `a.py`:
```python
# pylint: disable=missing-module-docstring,missing-function-docstring
# pylint: enable=consider-using-with


import contextlib


def read_files(file1, file2):
    with open(file1) as input_file1, (
        open(file2) if file2 else contextlib.nullcontext()
    ) as input_file2:
        if input_file2:
            return input_file1.read(), input_file2.read()
        return input_file1.read()
```

### Current behavior

```
t.py:10:8: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
```
-->

### Expected behavior

No warning.  I'm already using a `with` statement.

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.9.1
astroid 2.6.2
Python 3.8.10 (default, Jun  4 2021, 15:09:15)
[GCC 7.5.0]
```

**Comment by Pierre-Sassoulas** [comment]
Thanks for reporting the issue. It's probably due to the ``contextlib.nullcontext()``, and related to #4665

**Comment by Pierre-Sassoulas** [comment]
@DudeNr33 maybe I'm wrong and it's only the double open in with that is causing a problem ? What do you think ?

**Comment by DudeNr33** [comment]
Either that, or the ternary if-else on file2.
I will look at it. 
So many corner cases for this check I haven't thought of. 😀

**Comment by adam-azarchs** [comment]
Simpler reproducer confirms the issue is the ternary:
```python
# pylint: disable=missing-module-docstring,missing-function-docstring
# pylint: enable=consider-using-with


def read_files(file1, file2, which):
    with (open(file1) if which else open(file2)) as input_file:
        return input_file.read()
```
gives two errors:
```
t.py:6:10: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
t.py:6:36: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
```
This isn't that weird a corner case, as it's a bit awkward to use `with` in situations like that otherwise.

**Comment by Pierre-Sassoulas** [comment]
It will be fixed in 2.9.4 thanks to @DudeNr33 !
